### ISSUE TITLE
Location for BoxAnnotation on categorical axis returns ValueError

### ISSUE THREAD

**Issue by landge** [bokeh/bokeh#8482]
Versions:

|Name                   |Version                   |Build               |Channel
|-----------------|-------------------|-------------|---------
bokeh                     |  1.0.0                         |py36_1000   | conda-forge
python                    |  3.6.5                       |  1                  |  conda-forge
notebook                |  5.5.0                     |    py36_0       |   conda-forge

OS:              macOS Mojave 10.14
Browser:     Chrome -- Version 70.0.3538.110 (Official Build) (64-bit)

I would like to specify the location for a BoxAnnotation on a categorical axis
with help of the category name. 
Bokeh returns ValueError

```
from bokeh.io import output_notebook, show
from bokeh.models import BoxAnnotation
from bokeh.plotting import figure

output_notebook()

cats = ['cat_A','cat_B']

cat_plot = figure(y_range=cats, x_range=[1,2],  title='Cat')
cat_plot.add_layout(BoxAnnotation(top=['cat_B']))

show(cat_plot)

```

```
ValueError                                Traceback (most recent call last)
<ipython-input-4-a04e8c1ea9fe> in <module>()
      8 
      9 cat = figure(y_range=cats, x_range=[1,2], plot_width=200, plot_height=200, toolbar_location=None, title='Cat')
---> 10 cat.add_layout(BoxAnnotation(top=['cat_B']))
     11 show(cat)

~/miniconda3/lib/python3.6/site-packages/bokeh/model.py in __init__(self, **kwargs)
    259         self._document = None
    260         self._temp_document = None
--> 261         super(Model, self).__init__(**kwargs)
    262         default_theme.apply_to_model(self)
    263 

~/miniconda3/lib/python3.6/site-packages/bokeh/core/has_props.py in __init__(self, **properties)
    252 
    253         for name, value in properties.items():
--> 254             setattr(self, name, value)
    255 
    256     def __setattr__(self, name, value):

~/miniconda3/lib/python3.6/site-packages/bokeh/core/has_props.py in __setattr__(self, name, value)
    279 
    280         if name in props or (descriptor is not None and descriptor.fset is not None):
--> 281             super(HasProps, self).__setattr__(name, value)
    282         else:
    283             matches, text = difflib.get_close_matches(name.lower(), props), "similar"

~/miniconda3/lib/python3.6/site-packages/bokeh/core/property/descriptors.py in __set__(self, obj, value, setter)
    544             raise RuntimeError("%s.%s is a readonly property" % (obj.__class__.__name__, self.name))
    545 
--> 546         self._internal_set(obj, value, setter=setter)
    547 
    548     def __delete__(self, obj):

~/miniconda3/lib/python3.6/site-packages/bokeh/core/property/descriptors.py in _internal_set(self, obj, value, hint, setter)
    765 
    766         '''
--> 767         value = self.property.prepare_value(obj, self.name, value)
    768 
    769         old = self.__get__(obj, obj.__class__)

~/miniconda3/lib/python3.6/site-packages/bokeh/core/property/bases.py in prepare_value(self, obj_or_cls, name, value)
    326                     break
    327             else:
--> 328                 raise e
    329         else:
    330             value = self.transform(value)

~/miniconda3/lib/python3.6/site-packages/bokeh/core/property/bases.py in prepare_value(self, obj_or_cls, name, value)
    319         try:
    320             if validation_on():
--> 321                 self.validate(value)
    322         except ValueError as e:
    323             for tp, converter in self.alternatives:

~/miniconda3/lib/python3.6/site-packages/bokeh/core/property/either.py in validate(self, value, detail)
    115         if not (value is None or any(param.is_valid(value) for param in self.type_params)):
    116             msg = "" if not detail else "expected an element of either %s, got %r" % (nice_join(self.type_params), value)
--> 117             raise ValueError(msg)
    118 
    119     # TODO (bev) implement this

ValueError: expected an element of either Auto or NumberSpec(String, Dict(Enum('expr', 'field', 'value', 'transform'), Either(String, Instance(Transform), Instance(Expression), Float)), Float), got ['cat_B']

```

**Comment by gshpychka** [comment]
@mattpap I see the milestone was set to 1.1.1 but never included, what does that mean for this issue?

**Comment by mattpap** [comment]
This was fixed as a part of PR #12276.
